packages/vega-parser/src/parsers/guides/axis-config.js

```diff
@@ -1,12 +1,35 @@
 import {Bottom, Top} from './constants';
 import {extend} from 'vega-util';
+import { isSignal } from '../../util';
+import { xAxisExpr, axisOrientExpr } from './axis-util';
 
 export default function(spec, scope) {
   var config = scope.config,
       orient = spec.orient,
-      xy = (orient === Top || orient === Bottom) ? config.axisX : config.axisY,
-      or = config['axis' + orient[0].toUpperCase() + orient.slice(1)],
-      band = scope.scaleType(spec.scale) === 'band' && config.axisBand;
+      xy,
+      or,
+      band;
+
+  if (!isSignal(spec.orient)) {
```

Comment:
(Minor) Instead of using negation, wouldn't it be easier to read if we flip the if/else?

Reply:
I was thinking the not signal case is the more general / simpler so I placed it first, but I also agree that it is more difficult to read. Addressed.